substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ReachabilityHandlerFeature.java

```diff
@@ -88,7 +91,7 @@ private void registerReachabilityHandler(BeforeAnalysisAccess a, Object callback
         BeforeAnalysisAccessImpl access = (BeforeAnalysisAccessImpl) a;
         AnalysisMetaAccess metaAccess = access.getMetaAccess();
 
-        Set<Object> triggerSet = activeHandlers.computeIfAbsent(callback, c -> new HashSet<>());
+        ConcurrentHashMap.KeySetView<Object, Boolean> triggerSet = activeHandlers.computeIfAbsent(callback, c -> ConcurrentHashMap.newKeySet());
```

Comment:
that's actually the prototypical use case for `var triggerSet`, which we can now use in the native-image code base.

Reply:
Done.